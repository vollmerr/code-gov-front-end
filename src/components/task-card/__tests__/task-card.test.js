import React from 'react';

import TaskCardComponent from 'components/task-card'

import { shallow } from 'enzyme';


describe('Testing <TaskCardComponent /> component', () => {
  const props = {
    task: {
      languages: ['C#', 'HTML'],
      type: 'web',
      skill: '4',
      effort: '10',
      title: 'TestTitle',
      issueURL: 'testUrl',
      description: 'TestDescription',
      agency: {
        acronym: 'TestAcronym',
        name: 'TestName'
      }
    },
    date:{
      lastModified: '10/10/2010',
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TaskCardComponent {...props} />);
  });
  it('renders TaskCard', () => {
    expect(wrapper.exists()).toEqual(true);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should render correctly without description,title, or agencyName', () => {
    props.task.title = '';
    props.task.description = '';
    props.task.agency.name = '';
    wrapper = shallow(<TaskCardComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})