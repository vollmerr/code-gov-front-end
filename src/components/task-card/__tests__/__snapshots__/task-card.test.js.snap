// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TaskCardComponent /> component should render correctly 1`] = `
<div
  className="card-list-item card focusable"
>
  <h3
    className="card-title"
  >
    <a
      href="testUrl"
      target="_blank"
    >
      TestTitle
    </a>
  </h3>
  <p
    className="card-description"
  >
    TestDescription
  </p>
  <dl
    className="inline-after-600px show-w-gt-900"
  >
    <dt>
      Agency:
    </dt>
    <dd>
      <Connect(CustomLinkComponent)
        to="/browse-projects?agencies=TestAcronym"
      >
        TestName
      </Connect(CustomLinkComponent)>
    </dd>
    <CardPart
      text="Not Available"
      title="Last Updated"
    />
    <hr />
    <CardPart
      text="C#,HTML"
      title="Languages"
    />
    <CardPart
      text="Web"
      title="Type"
    />
    <CardPart
      text="4"
      title="Skill Level"
    />
    <CardPart
      text="10"
      title="Effort"
    />
  </dl>
</div>
`;

exports[`Testing <TaskCardComponent /> component should render correctly without description,title, or agencyName 1`] = `
<div
  className="card-list-item card focusable"
>
  <dl
    className="inline-after-600px show-w-gt-900"
  >
    <CardPart
      text="Not Available"
      title="Last Updated"
    />
    <hr />
    <CardPart
      text="C#,HTML"
      title="Languages"
    />
    <CardPart
      text="Web"
      title="Type"
    />
    <CardPart
      text="4"
      title="Skill Level"
    />
    <CardPart
      text="10"
      title="Effort"
    />
  </dl>
</div>
`;
